go/vt/vttablet/tabletmanager/action_agent.go

```diff
@@ -355,6 +356,16 @@ func NewActionAgent(
 			agent.initHealthCheck()
 		}()
 	} else {
+		// optionally populate metadata records
+		if *initPopulateMetadata {
+			// we can use agent.initialTablet here because it is set by Start()
```

Comment:
Couldn't we also just use `agent.Tablet()`? `InitTablet()` calls `setTablet()` if it succeeds. Is there any reason it has to be `initialTablet`?

Reply:
it doesn't have to be `initialTablet`. I noticed that `agent.Tablet()` always clones the tablet, and it seemed right to use `initialTablet` since we are still "initialiazing", though we have completed the `InitTablet` step.

Reply:
actually this has to be initialTablet. we set type to UNKNOWN on startingTablet, so the original type that is needed to populate metadata correctly has to be read from initialTablet.